src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/MessageConverterFactory.cs

```diff
@@ -1,42 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using System.Collections;
-using Microsoft.WindowsAzure.Storage.Queue;
-
-namespace Microsoft.Azure.WebJobs.Host.Queues.Bindings
-{
-    internal static class MessageConverterFactory
-    {
-        internal static IMessageConverterFactory<TInput> Create<TInput>()
-        {
-            if (typeof(TInput) == typeof(CloudQueueMessage))
-            {
-                return (IMessageConverterFactory<TInput>)new CloudQueueMessageConverterFactory();
-            }
-            else if (typeof(TInput) == typeof(string))
-            {
-                return (IMessageConverterFactory<TInput>)new StringMessageConverterFactory();
-            }
-            else if (typeof(TInput) == typeof(byte[]))
-            {
-                return (IMessageConverterFactory<TInput>)new ByteArrayMessageConverterFactory();
-            }
-            else
-            {
-                if (typeof(TInput).IsPrimitive)
-                {
-                    throw new NotSupportedException("Primitive types are not supported.");
```

Comment:
restrictions and exception type came from [Queue]. 

---
In reply to: [62360839](https://github.com/Azure/azure-webjobs-sdk/pull/711#discussion_r62360839) [](ancestors = 62360839)